lib/std/treap.zig

```diff
@@ -447,3 +455,194 @@ test "insert, find, replace, remove" {
         try testing.expectEqual(entry.node, treap.getEntryFor(key).node);
     }
 }
+
+test "inorderIterator" {
+    var treap = TestTreap{};
+    var nodes: [10]TestNode = undefined;
+
+    // Build the tree.
+    var i: usize = 0;
+    while (i < 10) : (i += 1) {
+        const key = @as(u64, i);
+        var entry = treap.getEntryFor(key);
+        entry.set(&nodes[i]);
+    }
+
+    // Test the iterator.
+    var iter = treap.inorderIterator();
+    i = 0;
+    while (iter.next()) |node| {
+        const key = @as(u64, i);
+        try testing.expectEqual(key, node.key);
+        i += 1;
+    }
+}
+
+test "getMin, getMax" {
+    var treap = TestTreap{};
+    var nodes: [3]TestNode = undefined;
+
+    try testing.expectEqual(treap.getMin(), null);
+    try testing.expectEqual(treap.getMax(), null);
+    { // nodes[1]
+        var entry = treap.getEntryFor(1);
+        entry.set(&nodes[1]);
+        try testing.expectEqual(treap.getMin(), &nodes[1]);
+        try testing.expectEqual(treap.getMax(), &nodes[1]);
+    }
+    { // nodes[0]
+        var entry = treap.getEntryFor(0);
+        entry.set(&nodes[0]);
+        try testing.expectEqual(treap.getMin(), &nodes[0]);
+        try testing.expectEqual(treap.getMax(), &nodes[1]);
+    }
+    { // nodes[2]
+        var entry = treap.getEntryFor(2);
+        entry.set(&nodes[2]);
+        try testing.expectEqual(treap.getMin(), &nodes[0]);
+        try testing.expectEqual(treap.getMax(), &nodes[2]);
+    }
+}
```

Comment:
How about a test something like this instead?

```zig
test "getMin, getMax" {
    var nodes: [100]TestNode = undefined;
    var prng = std.Random.DefaultPrng.init(0xdeadbeef);
    var iter = SliceIterRandomOrder(TestNode).init(&nodes, prng.random());

    var treap = TestTreap{};
    var min: u64 = std.math.maxInt(u64);
    var max: u64 = 0;

    try testing.expectEqual(null, treap.getMin());
    try testing.expectEqual(null, treap.getMax());

    // Insert and check min/max after each insertion.
    iter.reset();
    while (iter.next()) |node| {
        const key = prng.random().int(u64);

        // Insert into `treap`.
        var entry = treap.getEntryFor(key);
        entry.set(node);

        if (key < min) min = key;
        if (key > max) max = key;

        const min_node = treap.getMin().?;
        try std.testing.expectEqual(null, min_node.prev());
        try std.testing.expectEqual(min, min_node.key);

        const max_node = treap.getMax().?;
        try std.testing.expectEqual(null, max_node.next());
        try std.testing.expectEqual(max, max_node.key);
    }
}
```

That way it also tests that `getMin`/`getMax` will work for larger `Treap`s and it checks that `prev()` of `getMin()` is always `null` and the `next()` of `getMax()` is always `null`.

Reply:
I kept both tests.

My experiences with randomized tests are not very bright due to their behavior hard to fully reason... So would like to keep the simple one.